packages/eds-core-react/src/components/Datepicker/calendars/Calendar.tsx

```diff
@@ -0,0 +1,111 @@
+import { useCalendar, useLocale } from 'react-aria'
+import { forwardRef, ReactNode, RefObject, useEffect, useState } from 'react'
+import { CalendarStateOptions, useCalendarState } from '@react-stately/calendar'
+
+import { CalendarGrid } from './CalendarGrid'
+import { CalendarHeader } from './CalendarHeader'
+import { HeaderProps } from '../props'
+import { DatePickerState } from '@react-stately/datepicker'
+import {
+  CalendarDate,
+  CalendarDateTime,
+  createCalendar,
+  toCalendarDate,
+} from '@internationalized/date'
+import { Popover } from '../../Popover'
+
+/**
+ * Calendar is the inline calendar picker used in {@link DatePicker}.
+ * It only allows selecting a single date
+ */
+export const Calendar = forwardRef(
+  (
+    {
+      Header,
+      footer,
+      state,
+      ...props
+    }: {
+      /**
+       * Custom header component
+       */
+      Header?: (props: HeaderProps) => ReactNode
+      /**
+       * Custom footer component
+       */
+      footer?: ReactNode
+      /**
+       * The calendar state from react-aria
+       */
+      state: DatePickerState
+    } & CalendarStateOptions<CalendarDate | CalendarDateTime>,
+    ref,
+  ) => {
+    const [showYearPicker, setShowYearPicker] = useState(false)
+    const { locale } = useLocale()
+    const calendarState = useCalendarState({
+      ...props,
+      locale,
+      createCalendar,
+    })
+
+    const { calendarProps, prevButtonProps, nextButtonProps, title } =
+      useCalendar(props, calendarState)
+
+    useEffect(() => {
+      if (state.value) {
+        calendarState.setFocusedDate(toCalendarDate(state.value))
+      }
+      // We don't want to trigger an update when the state changes (spoiler alert: the state changes every time focus changes)
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [state.value])
```

Comment:
I like to put side effects like this at the top of the component so that they are really visible. Important when debugging because these side effects are often the root cause of bugs.

Reply:
Removed this effect :)